### PR TITLE
Check if any s3 bucket has bucket policy and if it does mark it non-compliant

### PR DESCRIPTION
**Pull request by droidlabour**

### CODE REVIEW

**Comment by droidlabour** [comment]
@gregjinkim I've renamed it to `s3_bucket_policy_prohibited.py` Good catch.
Also I've changed to use Configuration Item instead of boto3. Apparently I wasn't aware of it.

**Comment by droidlabour** [comment]
@gregjinkim Commit e7fae5d - 07e9088 were from my previous PR https://github.com/awslabs/aws-config-rules/pull/41/commits
I've added my latest commit on top of HEAD this repo.

File: python/s3_bucket_policy_prohibited.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import boto3
+import json
+import logging
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+
+def evaluate_compliance(configuration_item):
+    bucket_policy = configuration_item["supplementaryConfiguration"].get("BucketPolicy")
```
Comment (gregjinkim) [inline python/s3_bucket_policy_prohibited.py]: Can you handle cases of 
1) Configuration Items for resource types other than s3 buckets https://github.com/awslabs/aws-config-rules/blob/master/python/s3-exposed-bucket.py#L40
2) Configuration Items for deleted resources https://github.com/awslabs/aws-config-rules/blob/master/python/s3-exposed-bucket.py#L47
Both cases should be not_applicable

Comment (droidlabour) [reply python/s3_bucket_policy_prohibited.py]: Added in 214072f

**Comment by droidlabour** [comment]
@gregjinkim Thanks for merging. But It looks like I missed adding this new rule in the `RULES.md` doc. Can you please merge https://github.com/PriceBoardIn/aws-config-rules/commit/006708e59648a74c22cddcb11547ed7385d80cca

**Pull request merged**